test_util/helpers.py

```diff
@@ -25,9 +25,9 @@ def __init__(self, auth_json):
         self.auth_cookie = None
 
     def authenticate(self, cluster):
-        assert isinstance(cluster, test_util.cluster_api.ClusterApi), 'Unrecognized cluster object!'
         logging.info('Attempting authentication')
-        r = cluster.post(path='/acs/api/v1/auth/login', user=None, json=self.auth_json)
+        # Get a non-user session so that auth headers are absent
+        r = cluster.get_user_session(None).post(path='/acs/api/v1/auth/login', json=self.auth_json)
```

Comment:
get_user_session(None) creates a copy of the whole object every time, could we say `if user is none then .post() otherwise do the copy`?